gpumon/gpumon_server.ml

```diff
@@ -29,10 +29,23 @@ module Make(I: Interface) = struct
     | None -> raise Gpumon_interface.NvmlInterfaceNotAvailable
 
     let get_pgpu_metadata _ dbg pgpu_address =
+      let this = "get_pgpu_metadata" in
       let interface = get_interface_exn () in
       try
-        let device = Nvml.device_get_handle_by_pci_bus_id interface pgpu_address in
-        Nvml.device_get_pgpu_metadata interface device
+        let device =
+          Nvml.device_get_handle_by_pci_bus_id interface pgpu_address in
+        let compat =
+          Nvml.device_get_pgpu_metadata interface device in
+        let version, revision, driver =
+          ( Nvml.pgpu_metadata_get_pgpu_version compat
+          , Nvml.pgpu_metadata_get_pgpu_revision compat
+          , Nvml.pgpu_metadata_get_pgpu_host_driver_version compat
+          ) in
+        let major, minor =
+          Scanf.sscanf driver "%d.%d" (fun x y -> x, y) in
```

Comment:
For now it looks good. I'd maybe stay on the safe side and only scan `"%d."`, ignoring the minor. But this is just a minor comment, feel free to ignore it